platform/core.windows/src/org/netbeans/core/windows/model/DefaultModeModel.java

```diff
@@ -232,7 +233,23 @@ public int compare(TopComponent tc1, TopComponent tc2) {
     }
 
     private int compareFullFilePath(FileObject f1, FileObject f2) {
-        return FileUtil.toFile(f1).compareTo(FileUtil.toFile(f2));
+        return toFullFilePath(f1).compareToIgnoreCase(toFullFilePath(f2));
+    }
+
+    private String toFullFilePath(FileObject fo) {
+        File f = FileUtil.toFile(fo);
+
+        if (f != null) {
+            return f.getAbsolutePath();
+        }
+
+        FileObject rootFO = FileUtil.getArchiveFile(fo);
+
+        if (rootFO != null) {
+            return toFullFilePath(rootFO) + "/" + fo.getPath();
+        }
+
+        return fo.toURL().getPath();
```

Comment:
working with URLs can be expensive since it involves dns resolution which is blocking (never put them as keys in a map). Comparing Srings or URIs wouldn't have this issue.